src/js/files/components/UploadOverlay.tsx

```diff
@@ -0,0 +1,103 @@
+import { every, map, reject, reverse, sumBy } from "lodash-es";
+import React from "react";
+import { connect } from "react-redux";
+import styled from "styled-components";
+import { getFontSize, getFontWeight } from "../../app/theme";
+import { Badge, BoxGroup, BoxGroupHeader, BoxGroupSection } from "../../base";
+import { UploadItem } from "./UploadItem";
+import { UploadTime } from "./UploadTime";
+
+/**
+ * @prop show - whether the overlay should be displayed
+ */
+const StyledUploadOverlay = styled.div<{ show: boolean }>`
+    bottom: 0;
+    display: ${props => (props.show ? "block" : "none")};
+    max-width: 500px;
+    padding: 0 15px 15px 0;
+    position: fixed;
+    right: 0;
+    width: 35%;
+    z-index: 90;
+`;
+
+const UploadOverlayContent = styled(BoxGroup)`
+    background-color: ${props => props.theme.color.white};
+    box-shadow: ${props => props.theme.boxShadow.lg};
+    margin: 0;
+
+    ${BoxGroupHeader} {
+        align-items: auto;
+        display: block;
+        font-weight: ${getFontWeight("thick")};
+        font-size: ${getFontSize("lg")};
+    }
+`;
+
+const UploadOverlayList = styled(BoxGroupSection)`
+    height: auto;
+    max-height: 200px;
+    overflow-x: hidden;
+    padding: 0;
+`;
+
+const StyledUploadInformation = styled.div`
+    display: flex;
+    justify-content: space-between;
+    font-size: ${getFontSize("md")};
+    font-weight: ${getFontWeight("normal")};
+    margin-top: 5px;
+`;
+
+type UploadProps = {
+    fileType: string;
+    /* Unique identifier for the upload */
+    localId: string;
+    name: string;
+    /* Progress of the upload in percentage */
+    progress: number;
+    /* Size of the file in bytes */
+    size: number;
+    /* Remaining time in seconds */
+    remaining: number;
+    /* Upload speed in bytes per second */
+    uploadSpeed: number;
+};
```

Comment:
![Screenshot from 2024-05-10 15-15-38](https://github.com/virtool/virtool-ui/assets/125520621/e5821b11-4d26-4a86-a6ce-beb042f4651d)
Maybe @ReeceHoffmann can confirm but this is the return type I'm getting from uploads. `overallUploadSpeed` and `totalRemainingUploadTime` return 0 or undefined for me. Aside from this, i would probably define `Uploads` in `types.ts` and define a type `UploadOverlayProps` with `uploads: Uploads[]`

Reply:
You're right, good catch. I also notice that the values for the `remaining` and `uploadSpeed` props of `UploadTime` are both undefined causing the site to break when uploading a file. If we don't need these two props then the `UploadTime` component is obsolete. @ReeceHoffmann thoughts?

Reply:
Component works perfectly for me on main.

@raastinn Did you try using a large file or a throttled connection? If the files uploads too fast then it may never update the estimate.

@jasont7 If it is always undefined that makes me thing there may be a mistake in the refactor. component works perfectly on main. Try checking out to main to see if your still seeing a crash. The component is not obsolete as it has a use case for the end user, it may however be bugged.

Reply:
I think I might know what the problem is here - I was uploading a very small file that seems to not cause the estimate to update, thus not having real values for `remaining` and `uploadSpeed`. However, in the refactor these two props are required as `number` types (as can be seen in `Upload` in `types.ts`).